### PR TITLE
User profile & snapshot features

### PR DESCRIPTION
**Pull request by Jai-C**
Please test:

1. Login and logout functionality
2. Access control on pages (including going to login page when already logged in)
3. Registration (link accessible via login page)
4. Saving snapshots using the button
5. Listing user snapshots (only snapshots belonging to that user)
6. Selecting and visualising saved snapshots

### CODE REVIEW

**Comment by pproe** [comment]
Looks good to me! will wait for @hadeneh to review before merging

**Review by pproe — APPROVED** [review]

**Review by hadeneh — APPROVED** [review]

**Pull request merged**
